plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -1900,6 +1932,52 @@ protected function persist_order_to_db( &$order, bool $force_all_fields = false
 		$this->update_address_index_meta( $order, $changes );
 		$default_taxonomies = $this->init_default_taxonomies( $order, array() );
 		$this->set_custom_taxonomies( $order, $default_taxonomies );
+
+		if ( $order->has_cogs() && $this->cogs_is_enabled() ) {
+			$this->save_cogs_data( $order );
+		}
+	}
+
+	/**
+	 * Save the Cost of Goods Sold value of a given order to the database.
+	 *
+	 * @param WC_Abstract_Order $order The order to save the COGS value for.
+	 */
+	private function save_cogs_data( WC_Abstract_Order $order ) {
+		$cogs_value = $order->get_cogs_total_value();
+
+		/**
+		 * Filter to customize the Cost of Goods Sold value that gets saved for a given order,
+		 * or to suppress the saving of the value (so that custom storage can be used).
+		 *
+		 * @since 9.5.0
+		 *
+		 * @param float|null $cogs_value The value to be written to the database. If returned as null, nothing will be written.
+		 * @param WC_Abstract_Order $item The order for which the value is being saved.
+		 */
+		$cogs_value = apply_filters( 'woocommerce_save_order_cogs_value', $cogs_value, $order );
+		if ( is_null( $cogs_value ) ) {
+			return;
+		}
+
+		$existing_meta = $this->data_store_meta->get_metadata_by_key( $order, '_cogs_total_value' );
+
+		if ( 0.0 === $cogs_value ) {
+			if ( $existing_meta ) {
+				$existing_meta = current( $existing_meta );
+				$this->data_store_meta->delete_meta( $order, $existing_meta );
+			}
+		} elseif ( $existing_meta ) {
```

Comment:
I realize whether this is an improvement or not is subjective, so feel free to skip, but we could avoid nesting here:

```suggestion
		if ( 0.0 === $cogs_value && $existing_meta ) {
			$existing_meta = current( $existing_meta );
			$this->data_store_meta->delete_meta( $order, $existing_meta );
		} elseif ( $existing_meta ) {
```

Reply:
Indeed. The cleaner the code the better, so let's do it.